checker/match_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	. "github.com/onsi/gomega"
+	"testing"
+
+	"github.com/envoyproxy/data-plane-api/api/auth"
+
+	"github.com/projectcalico/app-policy/proto"
+)
+
+// Successful parse should return name and namespace.
+func TestParseSpiffeIdOk(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	id := "spiffe://foo.bar.com/ns/sandwich/sa/bacon"
+	name, namespace, err := parseSpiffeId(id)
+	g.Expect(name).To(Equal("bacon"))
+	g.Expect(namespace).To(Equal("sandwich"))
+	g.Expect(err).To(BeNil())
+}
+
+// Unsuccessful parse should return an error.
+func TestParseSpiffeIdFail(t *testing.T) {
```

Comment:
I think you can use Go sub-tests to group some of these together and run as sub-tests

Reply:
I didn't refactor that one, since it had just two cases and they use different `Expect` clauses, but did for the ones with 3 or more.